sensor/kubernetes/listener/resources/virtualmachine/dispatcher/virtualmachineinstances.go

```diff
@@ -0,0 +1,99 @@
+package dispatcher
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	virtualMachineV1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/component"
+	"github.com/stackrox/rox/sensor/kubernetes/listener/resources/virtualmachine/store"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeVirtV1 "kubevirt.io/api/core/v1"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+//go:generate mockgen-wrapper
+type virtualMachineStore interface {
+	Get(uid string) *store.VirtualMachineInfo
+	AddOrUpdateVirtualMachine(vm *store.VirtualMachineInfo)
+	AddOrUpdateVirtualMachineInstance(uid, namespace string, vsockCID *uint32, isRunning bool)
+	RemoveVirtualMachine(uid string)
+	RemoveVirtualMachineInstance(uid string)
+}
+
+type VirtualMachineInstanceDispatcher struct {
+	clusterID string
+	store     virtualMachineStore
+}
+
+func NewVirtualMachineInstanceDispatcher(clusterID string, store virtualMachineStore) *VirtualMachineInstanceDispatcher {
+	return &VirtualMachineInstanceDispatcher{
+		clusterID: clusterID,
+		store:     store,
+	}
+}
+
+func (d *VirtualMachineInstanceDispatcher) ProcessEvent(
+	obj interface{},
+	_ interface{},
+	action central.ResourceAction,
+) *component.ResourceEvent {
+	unstructuredObj, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		log.Errorf("not of type 'Unstructured': %T", obj)
+		return nil
+	}
+	virtualMachineInstance := &kubeVirtV1.VirtualMachineInstance{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.Object, virtualMachineInstance); err != nil {
+		log.Errorf("unable to convert 'Unstructured' to 'VirtualMachineInstance': %v", err)
+		return nil
+	}
+	if virtualMachineInstance.GetUID() == "" {
+		log.Errorf("convertion from unstructured failed: %v", obj)
+		return nil
+	}
+	if len(virtualMachineInstance.GetOwnerReferences()) != 1 {
```

Comment:
Could you add a link to doc that says it's always 1

Reply:
This part of the code has changed a little bit since we need to take into consideration ReplicaSets.